File: shipley_checks/src/Stage_4/test_unified_categorization.R
#!/usr/bin/env Rscript
# Quick test of unified categorization

cat("=== Testing Unified Categorization ===\n\n")

# Source unified taxonomy
source("shipley_checks/src/Stage_4/explanation/unified_taxonomy.R")

# Test some examples
test_cases <- list(
  list(name = "Aphis fabae", role = "herbivore"),
  list(name = "Platycheirus scutatus", role = "predator"),
  list(name = "Platycheirus scutatus", role = "pollinator"),
  list(name = "Bombus terrestris", role = "pollinator"),
  list(name = "Adalia bipunctata", role = "predator"),
  list(name = "Aceria fraxini", role = "herbivore")
)

for (test in test_cases) {
  category <- categorize_organism(test$name, test$role)
  cat(sprintf("%s (%s) → %s\n", test$name, test$role, category))
}

File: shipley_checks/src/Stage_4/guild_scorer_rust/src/bin/test_unified_categorization.rs
//! Test unified taxonomic categorization

use guild_scorer_rust::explanation::unified_taxonomy::{OrganismCategory, OrganismRole};

fn main() {
    println!("=== Testing Unified Categorization ===\n");

    let test_cases = vec![
        ("Aphis fabae", Some(OrganismRole::Herbivore)),
        ("Platycheirus scutatus", Some(OrganismRole::Predator)),
        ("Platycheirus scutatus", Some(OrganismRole::Pollinator)),
        ("Bombus terrestris", Some(OrganismRole::Pollinator)),
        ("Adalia bipunctata", Some(OrganismRole::Predator)),
        ("Aceria fraxini", Some(OrganismRole::Herbivore)),
        ("Apis mellifera", Some(OrganismRole::Pollinator)),
        ("Amara aenea", Some(OrganismRole::Predator)),
        ("Myzus persicae", Some(OrganismRole::Herbivore)),
        ("Cantharis fusca", Some(OrganismRole::Predator)),
    ];

    for (name, role) in test_cases {
        let category = OrganismCategory::from_name(name, role);
        let role_str = match role {
            Some(OrganismRole::Herbivore) => "herbivore",
            Some(OrganismRole::Predator) => "predator",
            Some(OrganismRole::Pollinator) => "pollinator",
            None => "unknown",
        };
        println!("{} ({}) → {}", name, role_str, category.display_name());
    }

    println!("\n✓ All categorization tests completed");
}
